pkg/controller/v1beta1/inferenceservice/components/predictor.go

```diff
@@ -168,3 +171,18 @@ func addBatcherContainerPort(container *v1.Container) {
 		}
 	}
 }
+
+func addAgentAnnotations(isvc *v1beta1.InferenceService, annotations map[string]string) bool {
+	if isvc.Spec.Predictor.GetImplementation().GetStorageUri() == nil {
```

Comment:
For custom predictor this can be true, we need additional check to exclude custom predictor case(`len(isvc.Spec.Predictor.PodSpec.Containers) != 0`).